### ISSUE TITLE
Integrate Arm64 test runs in Helix

### ISSUE THREAD

**Issue by gkhanna79** [dotnet/corefx#6901]

**Comment by mellinoe** [comment]
@danmosemsft @markwilkie Has this been done? Is there potentially a better place to track this issue?

**Comment by markwilkie** [comment]
As far as I know, this is not yet done.  However, it should be relatively straight forward to do.  Probably the right thing to do is talk to @Chrisboh about getting a working group going.  Timeframe goals?

Tracking eng issue: https://github.com/dotnet/core-eng/issues/120

**Comment by karelz** [comment]
@gkhanna79 should we move the issue to core-eng repo only? It does not seem to be specific to corefx repo.

**Comment by gkhanna79** [comment]
This is specific to running coreFX tests in Helix for WinArm64 (similar to what is done for x64 and x86). I am not sure what "core-eng" repo is about but we need to ensure this work is tracked at the appropriate source.

**Comment by Chrisboh** [comment]
There is things that need to happen here.  

1. We need to get the hardware to host in the lab and get that all setup.  I am not sure who is responsible for acquiring all the hardware but once we have it let me know and we can get that setup in the Lab.
2. We need to hook them up in Helix (https://github.com/dotnet/core-eng/issues/125)
3. We need the make sure the test builds support this new TFM and build and submit the test cases correctly


As you can see above I have added a new issue on our side to track the work that we will need to do.  Do you guys have a timeline on when we will have machines and builds to run?

**Comment by karelz** [comment]
> This is specific to running coreFX tests in Helix for WinArm64

I thought it is common across repos (coreclr, potentially others like wcf, CLI, etc.). Or what is so special on CoreFX regarding arm64?

**Comment by gkhanna79** [comment]
Currently, CoreFX is the only repo running tests in Helix. Also, the mechanism to run CoreFX tests (XUnit) is different from how CoreCLR tests are run (using smarty). 

I think it is appropriate to keep this issue active in CoreFX.

**Comment by karelz** [comment]
OK, that makes sense, thanks for clarification.

**Comment by ViktorHofer** [comment]
Are arm64 tests already running in Helix or still waiting for the hardware? If the hardware is the blocker I suggest moving it to a future milstone

**Comment by danmosemsft** [comment]
@Chrisboh

**Comment by Chrisboh** [comment]
The hardware just finished on-boarding into the DTAP lab.  All that is left is getting the clients setup and hooked up to Helix.

**Comment by ViktorHofer** [comment]
Any work to on the corefx side?

**Comment by ViktorHofer** [comment]
@Chrisboh Is this planned for 2.0?

**Comment by Chrisboh** [comment]
We have a limited set of hardware in place that is currently planned for UWP work.  Nothing is stopping other runs from happening on this hardware but it may take a bit longer than normal to complete.  @danmosemsft I don't think Arm64 was on the 2.0 matrix, should it be?

**Comment by danmosemsft** [comment]
@Chrisboh it's not. Samsung are presumably handling all arm testing for 2.0. @Petermarcu does that match your expectation?

**Comment by Petermarcu** [comment]
Correct that arm64 is not for .NET Core 2.0 but we as you can tell, @Chrisboh has been bringing up infra to make sure we can keep doing the bringup work. Samsung is doing work for Linux arm32.

This can go into Future for now since we don't need to actively track it. @gkhanna79 let me know if you think I'm missing anything.

**Comment by gkhanna79** [comment]
Sounds good to me.


Thanks,
Gaurav
________________________________

**Comment by sdmaclea** [comment]
@richlander @RussKeldorph @Petermarcu Has there been any progress in allowing arm64 to run through Helix.  Or is this something we can discuss offline.

I was looking at enabling arm64-linux corex runs in CI, but it is still blocked on helix as far as I can tell.

BTW I cannot see dotnet/core-eng must be private.

**Comment by Chrisboh** [comment]
Hi @sdmaclea,

Right now we are in the process of getting the new hardware that we ordered stood up in the lab and hooked up to Helix.  Once that is done we should be able to start sending a lot more work through that queue.  Right now there is only one machine left standing in that queue so until then I would hold off.  As for timing I hope to have it all ready to go in the next week or so.

**Comment by ViktorHofer** [comment]
On https://mc.dot.net/#/product/netcore/30/source/official~2Fcorefx~2Fmaster~2F/type/build~2Fproduct~2F/build/20180926.03/workItem/Orchestration/analysis/external/Link I can see that we have netcoreapp-arm64 official builds. 

@joshfree I believe we should add a uap-arm64 leg to the official builds. I already updated the uap dependencies to work with arm64: https://github.com/dotnet/arcade/pull/874.

**Comment by joshfree** [comment]
Thanks @ViktorHofer -- @tommcdon owns the uap space now;  @tommcdon what are your thoughts here?

**Comment by Petermarcu** [comment]
Folks are adding more devices to the lab as we speak and there should also be a test queue available in the next couple days to send our test to for arm64-linux.

**Comment by wfurt** [comment]
seems like we have both Linux and Windows ARM64. closing.
